app/modules/wallet/addresses/addresses.route.ts

```diff
@@ -0,0 +1,76 @@
+import * as restify from 'restify';
+import AddressController from './addresses.controller';
+import * as Joi from 'joi';
+import { IRoute, IRouteConfig, HttpMethods, AuthStrategies } from '../../../interfaces/utils/Route';
+import { oneLine } from 'common-tags';
+
+class WalletsRoute implements IRoute {
+  public basePath = '/api/wallets/:wallet/addresses';
+  public controller = new AddressController();
+  public swaggerTag = 'Wallet Address';
+
+  public getServerRoutes(): IRouteConfig[] {
+    return [
+      {
+        method: HttpMethods.GET,
+        auth: AuthStrategies.OAUTH,
+        handler: this.controller.get,
+        validation: {
+          schema: {
+            params: Joi.object().keys({
+              wallet: Joi.string().required(),
+            }).required()
+          }
+        },
+        swagger: {
+          summary: 'Get Wallet Address',
+          description: 'Gets addresses of given wallet',
+          responses: []
+        }
+      },
+      {
+        method: HttpMethods.PUT,
+        auth: AuthStrategies.OAUTH,
+        handler: this.controller.put,
+        validation: {
+          schema: {
+            params: Joi.object().keys({
+              wallet: Joi.string().required(),
+            }).required(),
+            body: Joi.object().keys({
+              chain: Joi.string().required(),
```

Comment:
You can add valid chains here. Like an enum.
```javascript
Joi.string.valid(['eth', 'btc']).required()
```

Reply:
Supporting only these two right now, or are there others as well?

Reply:
For now, these 2 yes.